services/app-web/src/s3/s3Amplify.ts

```diff
@@ -77,12 +77,49 @@ function newAmplifyS3Client(bucketConfig: S3BucketConfigType): S3ClientT {
             filename: string,
             bucket: BucketShortName
         ): Promise<void | S3Error> => {
+            // Construct the full key including the bucket prefix
+            const fullKey = `${bucketConfig[bucket]}/${filename}`
+            console.info(`Attempting to tag file as deleted: ${fullKey}`)
+            let metadata
+
             try {
-                await Storage.remove(filename, {
-                    bucket: bucketConfig[bucket],
-                })
+                const result = await Storage.getProperties(fullKey)
+                metadata = result.metadata
+                console.info('Successfully got file properties', metadata)
+            } catch (getPropertiesError) {
+                console.error('Error in getProperties:', getPropertiesError)
+                throw getPropertiesError
+            }
+
+            try {
+                // Add or update the 'deleted' tag
+                console.info('Preparing updated metadata')
+                const updatedMetadata = {
+                    ...metadata,
+                    deleted: 'true',
+                    deletedAt: new Date().toISOString(),
+                }
+
+                // Update the file's metadata
+                console.info('Copying file to update metadata')
+                try {
+                    await Storage.copy(
+                        { key: fullKey },
+                        { key: fullKey },
+                        { metadata: updatedMetadata }
+                    )
+                    console.info('Successfully updated file metadata')
```

Comment:
Interesting idea, I like it because we still retain something to show that a field may have been removed in the UI.